Lib/test/test_importlib/test_metadata_api.py

```diff
@@ -172,6 +172,11 @@ def test_entry_points_groups_get(self):
             entry_points().get('entries', 'default') == entry_points()['entries']
             entry_points().get('missing', ()) == ()
 
+    def test_entry_points_allows_no_attributes(self):
+        self.assertRaises(
+            AttributeError, setattr, entry_points()['entries'][0], 'foo', 4
+        )
```

Comment:
I personally think it's much more readable to use `assertRaises` as a context manager, e.g.

```suggestion
        with self.assertRaises(AttributeError):
            entry_points()['entries'][0].foo = 4
```

As a bonus, it's also more concise :)

Reply:
ah, didn't even know this was an option! It's been quite a few years since I've used `unittest` 😅 after switching to `pytest` 

with context manager solution is a clear improvement, will adjust.